.circleci/config.yml

```diff
@@ -118,6 +145,17 @@ workflows:
           filters:
             tags:
               only: /^v([0-9])+.([0-9])+.([0-9])+.*/
+      - publish-check:
+          requires:
+            - cross-compile
+            - loadtest
+            - test
+            - gosec
+            - coverage
+            - windows-test
```

Comment:
We can remove `loadtest`, `test`, `gosec`, `coverage`, and `windows-test` if they are not actually required for `publish-check`.

Reply:
It wouldn't solve anything other than may run the publish-check job in parallel. I think it's worth it having the publish-check job behave as close to the real publish job as possible. We could actually use the same publish job for check and add some env var to disable actual publishing but that'd just give us more bugs to deal with in future IMO.